src/vmm/src/lib.rs

```diff
@@ -299,6 +302,25 @@ impl Vmm {
         Ok(())
     }
 
+    /// Sends a pause command to the vCPUs.
+    pub fn pause_vcpus(&mut self) -> Result<()> {
+        for handle in self.vcpus_handles.iter() {
+            handle
+                .send_event(VcpuEvent::Pause)
+                .map_err(Error::VcpuEvent)?;
+        }
+        for handle in self.vcpus_handles.iter() {
+            match handle
+                .response_receiver()
+                .recv_timeout(Duration::from_millis(1000))
+            {
+                Ok(VcpuResponse::Paused) => (),
+                _ => return Err(Error::VcpuPause),
+            }
+        }
```

Comment:
nit: maybe this can go in a helper function `self.expect_vcpu_response(expected_response, error_otherwise)` that we can use everywhere to deduplicate code.

Reply:
I don't really like to give an error as parameter :(.

Reply:
`self.expect_vcpu_response(expected_response).ok_or(error)` looks better to you?

This is just a nit about code deduplication, treat it accordingly